test/simulator/timestamp/src/include/error_simulator.h

```diff
@@ -44,5 +44,24 @@
             return (__ret);     \
     } while (0)
 
+#define WT_TXN_SIM_RET(_txn_error, a) \
+    do {                              \
+        int __ret;                    \
+        if ((__ret = (a)) != 0) {     \
+            _txn_error = true;        \
+            return (__ret);           \
+        }                             \
+    } while (0)
+
+#define WT_TXN_SIM_RET_MSG(_txn_error, a, msg) \
```

Comment:
Would it work to do the following?
```
#define WT_TXN_SIM_RET_MSG(a, msg)
```
I don't see the point of passing the variable if we know it exists in the scope of the macro where it is used. It feels weird that we are changing its value too.

I am not sure why, but I am not fully convinced of this macro and the one above that both use the `_txn_error` variable. I would suggest this:

```
#define WT_TXN_SIM_RET(a, msg)                 \
    do {                                       \
        int __ret;                             \
        if ((__ret = (a)) != 0) {              \
            _txn_error = true;                 \
            if(!msg.empty())                   \
              std::cerr << msg << std::endl;   \
            return (__ret);                    \
        }                                      \
    } while (0)
```

Reply:
Ah that's true, we don't need to pass in `_txn_error`. I don't think combining the two would help with the readability in the simulator though.